items/actions: fix cutscene mesh swaps

lahm86 opened the pull request:
Resolves #4136.
Resolves #4142.
Resolves #4145.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

This corrects mesh swap behaviour in cutscenes for Lara so that status is more accurately tracked. The flip effects are used in the data as toggles, but the difference when drawing/undrawing in game is that Lara's gun status is used to track mesh status effectively. The mesh pointers are now compared in cutscenes in each toggle call.

lahm86 committed:
items/actions: fix cutscene mesh swaps

This corrects mesh swap behaviour in cutscenes for Lara so that status
is more accurately tracked. The flip effects are used in the data as
toggles, but the difference when drawing/undrawing in game is that
Lara's gun status is used to track mesh status effectively. The mesh
pointers are now compared in cutscenes in each toggle call.

Resolves #4136.  Resolves #4142.  Resolves #4145.

lahm86 commented:
@aredfan pointed out this is OG - I wonder if this will cause issues elsewhere then as I don't think there is an "undraw" flip effect.

rr- commented:
I think we can just monitor for new regressions – this looks like a very reasonable change.

lahm86 commented:
The effect is only used in cutscenes, so I think it means where you might normally see guns reholstered, they'll no longer be. But I think we could add a new effect quite easily to make it more realistic. I'm thinking Atlantis, GW and Diving Area - aredfan can keep me right though, that's off the top of my head.

rr- commented:
Sounds good!

aredfan commented:
> The effect is only used in cutscenes, so I think it means where you might normally see guns reholstered, they'll no longer be. But I think we could add a new effect quite easily to make it more realistic. I'm thinking Atlantis, GW and Diving Area - aredfan can keep me right though, that's off the top of my head.

Thank you for checking, I think you've listed all the relevant cutscenes. 👍 

There's another case worth looking into, it's the part when Lara changes into her swimsuit and her holsters are empty.

_____________

**Edit:** Things to check:

- [x] Atlantis cutscene
- [x] TGW cutscene: does Lara have her pistols holstered at the end? (TR2X bug since 0.9)
- [x] DA cutscene: does Lara have her pistols when she changes into a swimsuit? (OG bug)
- [x] DA cutscene: does Lara have her pistols holstered at the end? (OG bug)

lahm86 commented:
I've updated the description above and this should now fix all the snags with mesh swaps in cutscenes. I've also fixed the textures in DA cutscene as those were very jarring. Ref: https://github.com/LostArtefacts/TRXInjectionTool/pull/142

aredfan reviewed (approved):
Thank you very much @lahm86! 😌
